src/ripple/protocol/Serializer.h

```diff
@@ -298,7 +297,7 @@ class Serializer
     }
 
     // low-level VL length encode/decode functions
-    static Blob encodeVL (int length);
+    static int encodeVL (int length, std::array<std::uint8_t, 4>& result);
```

Comment:
@scottschurr I prefer your std::pair return. Splitting the return type into an output param and returning the actual length is a little weird. Will change. Fixing the memory allocation was a win, I'm not concerned about the cost of return a four byte array (and RVO makes this a non-concern, as you say).